docs/source/new-docs/tutorials/2024-tutorial.rst

```diff
@@ -0,0 +1,515 @@
+   functionality within GeoIPS. For example, some commonly used interfaces include the
+   ``algorithms``, ``colormappers``, and ``sectors`` interfaces.
+
+#. Family
+
+   A ``family`` is a subset of an interface's plugins which accept specific sets of
+   arguments/properties. Module-based plugins of the same ``family`` have similar call
+   signatures. YAML-based plugins of the same ``family`` are validated against the same
+   schema (i.e. they contain the same properties).
+
+#. Docstring
+
+   A ``docstring`` is a chunk of documentation which describes what your plugin does. This
+   property is required for every GeoIPS plugin created, module-based or YAML-based. We
+   require this property for proper documentation of created plugins, and it will also be
+   a useful feature later on when the GeoIPS Command Line Interface (CLI) is created, as
+   you will be able to see what each plugin does provided the ``docstring`` for that plugin
+   is filled.
+
+.. _plugin-development-setup:
+
+Product Plugin Development Initial Setup
+****************************************
+
+Before creating a new product for CLAVR-x Cloud-Top-Height, let's get the initial setup done:
+
+#. To develop a new GeoIPS plugin, install :ref:`GeoIPS<linux-installation>` and make sure that
+   you have ``geoips`` Python environment enabled throughout this tutorial using
+
+   .. code-block:: shell
+
+    mamba activate geoips # activating python envrionment
+
+   You will know if geoips environment is enabled if it shows up ahead of your username in your command prompt.
+
+#. Next, let's install GeoIPS CLAVR-x package and test the installation. This is needed as we are developing products
+for GeoIPS CLAVR-x.
+
+   .. code-block:: shell
+
+    git clone https://github.com/NRLMMD-GEOIPS/geoips_clavrx $GEOIPS_PACKAGES_DIR/geoips_clavrx # download the remote
+    repository
+    pip install -e $GEOIPS_PACKAGES_DIR/geoips_clavrx # installing the geoips_clavrx
+
+    $GEOIPS_PACKAGES_DIR/geoips/setup/check_system_requirements.sh test_data test_data_clavrx      # Install the clavrx
+    test data repo
+    $GEOIPS_PACKAGES_DIR/geoips_clavrx/tests/test_all.sh  # Run tests to verify geoips-clavrx installation
+
+#. Now, set the following additonal environment variables which are specific to your product plugin development
+
+   .. code-block:: shell
+
+      export MY_PKG_NAME=<your package name>    #read the note below for your package name,
+      export MY_PKG_DIR=$GEOIPS_PACKAGES_DIR/$MY_PKG_NAME    #your package directory
+      export MY_PKG_URL=<your package’s URL on version control platform(GitLab)> #your package VCS url
+
+   .. NOTE::
+    Choose a name for your package making sure that it is in lower case, starting with a letter,
+    and only contains letters, numbers, and underscores.
+
+#. Navigate to your product plugin directory and clone the example repository of customized plugin development,
```

Comment:
Please fix the grammar in this sentence. It is a run-on sentence, and has an invalid structure. For example, it could be:

`Navigate to your product plugin directory and clone the example repository of customized plugin development. The Template Basic Plugin repo is a base for building new plugin package that contain one or more custom plugins.`